sys/net/application_layer/gcoap/gcoap.c

```diff
@@ -399,7 +399,15 @@ static int _find_resource(coap_pkt_t *pdu, const coap_resource_t **resource_ptr,
                 resource++;
             }
 
-            int res = strcmp((char *)&uri[0], resource->path);
+            int res;
+            if ((resource->methods & COAP_MATCH_SUBTREE)) {
+                int len = strlen(resource->path);
+                res = memcmp(resource->path, uri, len);
```

Comment:
need to strncmp, otherwise, if len is > strlen(path), it would still compare that.

Reply:
You're right, changed!